be/src/olap/rowset/segment_v2/segment.cpp

```diff
@@ -604,6 +604,24 @@ vectorized::DataTypePtr Segment::get_data_type_of(const ColumnIdentifier& identi
     return nullptr;
 }
```

Comment:
please add test case for your improvement, maybe you could use DebugPoint to check the case